include/pybind11/buffer_info.h

```diff
@@ -79,6 +72,18 @@ struct buffer_info {
     }
 
 private:
+    struct private_ctr_tag { explicit private_ctr_tag() = default; };
```

Comment:
Is there any pointer in declaring the default constructor here?  (I.e. instead of just `struct private_ctr_tag {};`?)